caproto/_data.py

```diff
@@ -1,17 +1,107 @@
 import time
+from curio.meta import awaitable
```

Comment:
Does this introduce a hard curio dependency for the whole library? If so, I would just vendor `awaitable` in caproto itself.